File: identhub/src/main/java/de/solarisbank/identhub/di/LibraryModuleContextFactory.java
package de.solarisbank.identhub.di;

import android.content.Context;

import de.solarisbank.identhub.di.internal.Factory;
import de.solarisbank.identhub.di.internal.Preconditions;

public final class LibraryModuleContextFactory implements Factory<Context> {

    private final LibraryModule libraryModule;

    public LibraryModuleContextFactory(LibraryModule libraryModule) {
        this.libraryModule = libraryModule;
    }

    public static LibraryModuleContextFactory create(LibraryModule libraryModule) {
        return new LibraryModuleContextFactory(libraryModule);
    }

    @Override
    public Context get() {
        return Preconditions.checkNotNull(
                libraryModule.provideApplicationContext(),
                "Cannot return null from provider method"
        );
    }
}

File: core/src/main/java/de/solarisbank/sdk/core/di/LibraryModuleContextFactory.kt
package de.solarisbank.sdk.core.di

import android.content.Context
import de.solarisbank.sdk.core.di.internal.Factory
import de.solarisbank.sdk.core.di.internal.Preconditions

class LibraryModuleContextFactory(private val libraryModule: LibraryModule) : Factory<Context> {
    override fun get(): Context {
        return Preconditions.checkNotNull(
                libraryModule.provideApplicationContext(),
                "Cannot return null from provider method"
        )
    }

    companion object {
        @JvmStatic
        fun create(libraryModule: LibraryModule): LibraryModuleContextFactory {
            return LibraryModuleContextFactory(libraryModule)
        }
    }
}